src/components/collapsible-panel.tsx

```diff
@@ -0,0 +1,43 @@
+import { ToolbarButtonComponent } from '@jupyterlab/apputils';
+import React, { useState } from 'react';
+import { caretDownIcon, caretRightIcon } from '@jupyterlab/ui-components';
+
+export interface ICollapsiblePanelProps {
+  expandedByDefault?: boolean;
+  title: JSX.Element | string;
+  content: JSX.Element | string;
+}
+
+export function CollapsiblePanel(props: ICollapsiblePanelProps): JSX.Element {
+  const [expanded, setExpanded] = useState(props.expandedByDefault ?? false);
+
+  const onControlClick = () => {
+    setExpanded(!expanded);
+  };
+
+  return (
+    <div className={'jp-collapsible-panel' + (expanded ? ' expanded' : '')}>
+      <div
+        className={
+          'jp-collapsible-panel-summary' + (expanded ? ' expanded' : '')
+        }
+      >
+        <ToolbarButtonComponent
+          className="jp-collapsible-panel-icon-toolbar-button"
+          icon={expanded ? caretDownIcon : caretRightIcon}
+          onClick={onControlClick}
+        />
+        <span className="jp-collapsible-panel-title" onClick={onControlClick}>
+          {props.title}
+        </span>
+      </div>
+      <div
+        className={
+          'jp-collapsible-panel-content' + (expanded ? ' expanded' : '')
+        }
+      >
+        {expanded && <div>{props.content}</div>}
```

Comment:
Wouldn't this remove the content rather than hiding it? Is that the intended effect here?

Reply:
Agreed — if the content isn't visible, it might be better to leave it in the DOM so that other functions can modify it.

Reply:
That's a good point, thank you both for pointing this out. Changed to hiding content when panel is closed via css `display: none;`